Mod2/Lessons/Week5/DatabaseMigrations.md

```diff
@@ -0,0 +1,135 @@
+
+<p align='center'>
+  <img src='../../Images/Week5/migration_up_and_down.png'>
+</p>
+
+❓Do we want to store our migration files in Github?
+
+#### Code Organization
+
+As our projects grow, we will get to a point where we want more organization in our project file structure. The convention when working with entity framework is to have a folder called `Models` and a folder called `Data`. 
+
+Let's add those folders together and move our files into the appropriate folders. You will notice that Visual Studio suggests modifying the namespace, let's accept this suggestion. So the namespace for our models will be `PlantApp.Models` instead of just `PlantApp`.
+
+<p align='center'>
+  <img src='../../Images/Week5/folder_structure.png'>
+</p>
+
+### Seed Data
+
+Database seeding is populating a database with an initial set of data. Oftentimes you want to start your database with some initial data so that as you're testing edit or delete functionality you have a set of data to experiment with. It's also common to start your users table with information for all of the company's employees so they don't need to be added manually.
+
+There are many ways to create seed data. The approach we're going to use is building a DataSeeder Class similar to the following. Notice the `PlantApp.Data` namespace because this class belongs in the `Data` folder.
+
+```C#
+namespace PlantApp.Data
+{
+    public class DataSeeder
+    {
+        public static void SeedPlantsAndRooms(PlantTrackerContext context)
+        {
+            if (!context.Rooms.Any())
+            {
+                var fern_plant = new Plant {Type = "Fern", PurchaseDate = DateTime.Parse("1975-06-15T13:45:30-07:00").ToUniversalTime()};
+                var rubber_plant = new Plant {Type = "Rubber", PurchaseDate = DateTime.Parse("2021-01-15T11:45:02-07:00").ToUniversalTime()};
+                var jade_plant = new Plant {Type = "Jade", PurchaseDate = DateTime.Parse("2021-01-15T11:45:02-07:00").ToUniversalTime()};
+
+                var rooms = new List<Room>
+                {
+                 new Room
+                {
+                    Name = "Office",
+                    HasSunlight = true,
+                    Plants = new List<Plant> {fern_plant, rubber_plant}
+                },
+                new Room
+                {
+                    Name = "Kitchen",
+                    HasSunlight = true,
+                    Plants = new List<Plant> {jade_plant}
+                }
+            };
+                context.Rooms.AddRange(rooms);
+                context.SaveChanges();
+            }
+        }
+    }
+}
+```
+
+<!-- TODO: this might change based on the structure from the CRUD lesson -->
+
+When we want to run this file and seed some plants and rooms into our database we can add (or uncomment) the following in our `Program.cs` file.
+
+```C#
+using (var context = new PlantTrackerContext())
```

Comment:
Is this the first time we've seen `using` like this?  Should we provide more context on exactly where in the file this should go?

Reply:
I want to see how the CRUD lesson gets a context to use and will follow that same pattern here. I'm imagining it won't be anything new, but not sure who Richard is structuring that yet.

Reply:
Just took a look at Richard's PR and looks like he will already introduce using `(var context = new PlantTrackerContext())` so I imaging students will just add the line about running the seed data right below this line and comment it in or out.